core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/ClusterBootstrapper.java

```diff
@@ -359,6 +360,17 @@ public void bootstrapNewNodes(Long stackId, Set<String> upscaleCandidateAddresse
         }
     }
 
+    private void cleanupOldSaltState(List<GatewayConfig> allGatewayConfigs, Set<Node> nodes) throws CloudbreakOrchestratorFailedException {
+        List<GatewayConfig> gatewayTargets = allGatewayConfigs.stream()
+                .filter(gc -> nodes.stream().anyMatch(n -> gc.getPrivateAddress().equals(n.getPrivateIp())))
```

Comment:
Can we simplify this logic to _.filter(gc -> nodes.stream().anyMatch(n -> gc.getPrivateAddress().equals(n.getPrivateIp())))_ ? Then we can delete these lines: 
`List<GatewayConfig> saltMastersToCorrect = new ArrayList<>(allGatewayConfigs);
 saltMastersToCorrect.removeAll(gatewayTargets);`